src/lib.rs

```diff
@@ -0,0 +1,206 @@
+//! Tensor wrapper that exposes the `ndarray` API.
+//!
+//! This crate provides a small wrapper around the `Tensor` data
+//! structure of the `tensorflow` crate, to make it possible to use
+//! the `ndarray` API. This wrapper, `TensorArray`, provides the
+//! `view` and `view_mut` methods to respectively obtain `ArrayView`
+//! and `ArrayViewMut` instances.
+//!
+//! The following example shows how to wrap a `Tensor` and obtain
+//! an `ArrayView`:
+//!
+//! ~~~
+//! use ndarray::{arr2, Ix2};
+//! use ndarray_tensorflow::TensorArray;
+//! use tensorflow::Tensor;
+//!
+//! let tensor = Tensor::new(&[2, 3])
+//!     .with_values(&[0u32, 1, 2, 3, 4, 5])
+//!     .unwrap();
+//! let array: TensorArray<_, Ix2> =
+//!     TensorArray::from_tensor(tensor)
+//!     .unwrap();
+//! assert_eq!(array.view(),
+//!     arr2(&[[0, 1, 2], [3, 4, 5]]));
+//! ~~~
+
+use std::convert::TryFrom;
+use std::error::Error;
+use std::fmt;
+
+use ndarray::{ArrayView, ArrayViewMut, Dimension, IntoDimension};
+use tensorflow::{Tensor, TensorType};
+
+/// Mismatch between the tensor shape dimensionality and shape type.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ShapeError;
+
+impl fmt::Display for ShapeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "Mismatch between the tensor shape dimensionality and the shape type"
+        )
+    }
+}
+
+impl Error for ShapeError {}
+
+/// A wrapper for `Tensor` that provides views.
+///
+/// A Tensorflow `Tensor` only provides a limited API. This type is a
+/// wrapper around `Tensor` that makes it possible to use a tensor as
+/// an `ArrayView` or `ArrayViewMut` from the `ndarray` crate.
+pub struct TensorArray<T, D>
```

Comment:
That's a good point! I don't like `ArrayTensor`, since it suggest that it is a tensor of an array (which is weird, in contrast to `TensorView`).

What do you think of `NdTensor`?

Reply:
`NdTensor` is a bit redundant but given that the rust `Tensor` is 1d, it's probably the most reasonable thing to go with.

Reply:
One more idea: `TFArray`

Reply:
`TidyTensor`, `StructuredTensor`?

I'm still leaning towards `NdTensor`

Reply:
`ActualTensor` or `ProperTensor` because `tensorflow::Tensor` is just a 1-d array ;) .

Reply:
What I don't like is that the tensor variations of `ndarray` all have `Array` suffixes: `Array`, `RcArray`, `ArcArray`. This is what `TensorArray` come from. So, I am still leaning towards `TFArray` or `TensorflowArray`.